src/legacy/core_plugins/timelion/public/vis/index.ts

```diff
@@ -17,23 +17,19 @@
  * under the License.
  */
 
+// @ts-ignore
 import { VisFactoryProvider } from 'ui/vis/vis_factory';
 import { i18n } from '@kbn/i18n';
-import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
-import { TimelionRequestHandlerProvider } from './timelion_request_handler';
+// @ts-ignore
 import { DefaultEditorSize } from 'ui/vis/editor_size';
-
 // we also need to load the controller and directive used by the template
 import './timelion_vis_controller';
 import '../directives/timelion_expression_input';
-
+import { TimelionRequestHandlerProvider } from './timelion_request_handler';
 import visConfigTemplate from './timelion_vis.html';
 import editorConfigTemplate from './timelion_vis_params.html';
 
-// register the provider with the visTypes registry so that other know it exists
-VisTypesRegistryProvider.register(TimelionVisProvider);
-
-export default function TimelionVisProvider(Private) {
+export function TimelionVisProvider(Private: any) {
```

Comment:
Could use `IPrivate`, or even better shim this as we've done in other `vis_type*` plugins (`vis_type_vega` is a good example of how to get rid of `Private`)

Reply:
Currently `IPrivate` is added. The Shim's approach has been solving.